p2p/tpu.md

```diff
@@ -0,0 +1,234 @@
+Incoming QUIC traffic indicating IP fragmentation may be ignored.
+
+**Example QUIC frames**
+
+_First UDP packet_
+
+```
+STREAM Frame {
+  Prefix (5) = 0b00001
+  Offset Present (1) = 0   # Offset is absent for first fragment
+  Length Present (1) = 0   # Length is absent, as fragment frames always fill packet
+  Fin (1) = 0
+
+  Stream ID (i)
+  Stream Data (..)
+}
+```
+
+_Second UDP packet_
+
+```
+STREAM Frame {
+  Prefix (5) = 0b00001
+  Offset Present (1) = 1
+  Length Present (1) = 0
+  Fin (1) = 0
+
+  Stream ID (i)
+  [Offset (i)]
+  Stream Data (..)
+}
+```
+
+_Last UDP packet_
+
+```
+STREAM Frame {
+  Prefix (5) = 0b00001
+  Offset Present (1) = 1
+  Length Present (1) = 0
+  Fin (1) = 1             # Terminate stream
+
+  Stream ID (i)
+  [Offset (i)]
+  Stream Data (..)
+}
+```
+
+#### Acknowledgement
+	
+`STREAM` frames sent by clients are ack-eliciting.
+Refer to [RFC 9000 Section 13.2.1] for obligations on sending `ACK` frames.
+
+It is permitted to coalesce multiple `ACK` frames to reduce packet count.
+
+#### Stream Limits
+
+The server should send [`MAX_STREAM_DATA`] frames for recently created fragmented tx streams.
+The client should expect arbitrarily receive quotas for any stream.
+
+Note that stream IDs are incremental (ignoring the two least significant bits).
+Upon opening a stream with an out-of-order ID, all preceding streams are also created.
+
+In this case, the server may proactively send multiple [`MAX_STREAM_DATA`] frames for implicitly created streams.
+
+  [RFC 9000 Section 13.2.1]: https://www.rfc-editor.org/rfc/rfc9000.html#name-sending-ack-frames
+  [`MAX_STREAM_DATA`]: https://www.rfc-editor.org/rfc/rfc9000.html#name-max_stream_data-frames
+
+#### Aborting
+
+The server must terminate the QUIC connection in case of the following events:
+
+  - Submission of invalid transactions, specifically:
+    - Txs that fail deserialization
+    - Txs that fail parameter sanitization
+    - Txs that fail signature verification
+  - Transmission of QUIC datagrams, defined by [RFC 9221]
+  - Attempted creation of QUIC bidirectional streams (already forbidden by quota)
+  - All other cases where the QUIC RFC dictates connection termination.
+
+  [RFC 9221]: https://www.rfc-editor.org/rfc/rfc9221.html
+
+## TPU/UDP protocol
+
+TPU/UDP is a simple datagram-oriented network protocol.
+
+Each datagram carries a single serialized transaction without headers or padding.
+
+It is being deprecated in favor of TPU/QUIC, which adds authentication, confidentiality, and congestion control.
+
+## Security
+
+### Packet flood incentivization
+
+When TPU/UDP links become congested, packets start to get dropped arbitrarily.
+This incentivizes clients to repeatedly send packets (spam) to increase the chance
+of getting transactions confirmed. Such increase in traffic worsens congestion.
+
+TPU/QUIC implements retransmit and congestion control to address this vulnerability.
+
+### Confidentiality
```

Comment:
not sure if relevant here, but mainly due to what we're working on :)

Reply:
@buffalu: @segfaultdoc also called relevancy of MEV in this context into question: https://github.com/solana-foundation/specs/pull/1#discussion_r998460341
Resolving this thread as a duplicate but it's definitely relevant to bring up the scope of spec